elfi/testbench/testbench.py

```diff
@@ -0,0 +1,295 @@
+"""This module implements testbench-functionality in elfi."""
+
+import logging
+
+import numpy as np
+
+from elfi.visualization.visualization import ProgressBar
+
+logger = logging.getLogger(__name__)
+
+__all__ = ['Testbench', 'TestbenchMethod']
+
+
+class Testbench:
+    """Base class for comparing the performance of LFI-methods.
+
+       One elfi.Model can be inferred `repetitions`-times with
+       each of the methods included in `method_list`.
+
+    Attributes
+    ----------
+    model : elfi.Model
+        elfi.Model which is inferred.
+    method_list : list
+        List of elfi-infernce methods.
+    repetitions : int
+        How many repetitions of models is included in the testbench.
+    seed : int, optional
+
+
+    """
+
+    def __init__(self,
+                 model=None,
+                 repetitions=1,
+                 observations=None,
+                 reference_parameter=None,
+                 reference_posterior=None,
+                 progress_bar=True,
+                 seed=None):
+        """Construct the testbench object.
+
+        Parameters
+        ----------
+        model : elfi.Model
+            elfi.Model which is inferred. Needs to have get_model-method.
+        repetitions : int
+            How many repetitions of models is included in the testbench.
+        observation : np.array, optional
+            Observation, if available.
+        reference_parameter : dictionary, optional
+            True parameter values if available.
+        reference_posterior : np.array, optional
+            A sample from a reference posterior.
+        progress_bar : boolean
+            Indicate whether to display testbench progressbar.
+        seed : int, optional
+
+        """
+        self.model = model
+        self.method_list = []
+        self.method_seed_list = []
+        self.repetitions = repetitions
+        self.rng = np.random.RandomState(seed)
+        self.observations = observations
+        self.reference_parameter = reference_parameter
```

Comment:
would propose `self.reference_parameter = reference_parameter.copy()` because the dict contents is altered in `_resolve_reference_parameters`

Reply:
Modified as suggested. Similarly, modified `self.observations = observations.copy()`.